pkg/apiclient/workflowtemplate/workflow-template.proto

```diff
@@ -24,7 +24,8 @@ message WorkflowTemplateGetRequest {
 
 message WorkflowTemplateListRequest {
   string namespace = 1;
-  k8s.io.apimachinery.pkg.apis.meta.v1.ListOptions listOptions = 2;
+  string namePattern = 2;
+  k8s.io.apimachinery.pkg.apis.meta.v1.ListOptions listOptions = 3;
```

Comment:
changing the tag numbers is apparently a breaking change in protobufs, see https://github.com/argoproj/argo-workflows/pull/13160#discussion_r1634092491 for more details.